CMakeLists.txt

```diff
@@ -239,6 +250,80 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(hardened_cxx_flags "")
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Why is this gated in a COMPILER = "GNU" check (we don't do this in master)? I understand that currently only GCC might understand this flag, but I do not like the idea of doing this for anything hardening related. Just try and use the flag, and ignore if it doesn't work.

Reply:
https://github.com/bitcoin/bitcoin/pull/28672.

Reply:
Thanks! Implemented approach from https://github.com/bitcoin/bitcoin/pull/28672,

> Why is this gated in a COMPILER = "GNU" check (we don't do this in master)?

This check implements in the code what is implied by the comment in the master branch.